### PR TITLE
feat(sdk-crash-detection): Detect Cocoa Crashes

### PR DESCRIPTION
**Pull request by philippsuess**
This PR adds the implementation for the Cocoa SDK crash detection.

This is the third PR of splitting up the POC for SDK crash detection (https://github.com/getsentry/sentry/pull/49928) into multiple PRs.

### CODE REVIEW

File: src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py [hunk 1]
```diff
@@ -1,9 +1,45 @@
...
+
+        if function is not None:
+            # [SentrySDK crash] is a testing function causing a crash.
+            # Therefore, we don't want to mark it a as a SDK crash.
+            if "SentrySDK crash" in function:
+                return False
+
+            function_matchers = ["*sentrycrash*", "**[[]Sentry*"]
+            for matcher in function_matchers:
+                if glob_match(frame.get("function"), matcher, ignorecase=True):
```
Comment (iker-barriocanal) [inline src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py:35]:
```suggestion
                if glob_match(function, matcher, ignorecase=True):
```

File: src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py [hunk 2]
```diff
@@ -1,9 +1,45 @@
...
+                    return True
+
+        filename = frame.get("filename")
+        if filename is not None:
+            filenameMatchers = ["Sentry**"]
+            for matcher in filenameMatchers:
+                if glob_match(filename, matcher, ignorecase=True):
+                    return True
```
Comment (iker-barriocanal) [inline src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py:43]: Not sure if indentations is correct:
```suggestion
        if filename and glob_match(filename, "Sentry**", ignorecase=True):
                return True
```

File: src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py [hunk 3]
```diff
@@ -1,9 +1,45 @@
...
+
+            if frame.get("in_app") is True:
+                return False
+
+        return False
+
+    def is_sdk_frame(self, frame: Mapping[str, Any]) -> bool:
+        function = frame.get("function")
+
+        if function is not None:
```
Comment (iker-barriocanal) [inline src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py:27]:
```suggestion
        if function:
```

**Review by iker-barriocanal — APPROVED** [review]

File: src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py (hunk 2 above)
Comment (philipphofmann) [reply src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py:43]: No, I think it's correct.

File: src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py (hunk 3 above)
Comment (philipphofmann) [reply src/sentry/utils/sdk_crashes/cocoa_sdk_crash_detector.py:27]: TIL 👏

**Pull request merged**